allennlp/training/trainer.py

```diff
@@ -32,7 +32,8 @@ def __init__(self,
                  cuda_device: int = -1,
                  grad_norm: Optional[float] = None,
                  grad_clipping: Optional[float] = None,
-                 log_one_line_per_batch: bool = False) -> None:
+                 log_one_line_per_batch: bool = False,
+                 validation_metric_decreases: bool = True) -> None:
```

Comment:
I'm actually a little worried about someone specifying "overall-f1" and thinking things will just work, but forgetting to specify "validation_metric_decreases" to `False`.  I kind of think that if we have this additional parameter, it should be required, but it seems really easy to miss.  An alternative: require all metric strings to begin with either "-" or "+" (just in the `__init__` parameter), and remove it and set this flag in `__init__`.  It's explicit, and it requires to change the flag if you change the default validation metric.